runtime/lua/vim/lsp/tagfunc.lua

```diff
@@ -0,0 +1,69 @@
+local M = {}
+local lsp = vim.lsp
+
+local function mk_tag_item(name, range, uri)
+  local start = range.start
+  return {
+    name = name,
+    filename = vim.uri_to_fname(uri),
+    cmd = string.format(
+      'call cursor(%d, %d)', start.line + 1, start.character + 1
+    )
+  }
+end
+
+local function query_definition(pattern)
+  local params = lsp.util.make_position_params()
+  local results_by_client, err = lsp.buf_request_sync(0, 'textDocument/definition', params, 1000)
+  assert(not err, vim.inspect(err))
```

Comment:
On `err` return an empty table (e.g. if timeout expired) so that `tagfunc` will fallback to tags.